Source/CarthageKit/Xcode.swift

```diff
@@ -508,10 +509,23 @@ public struct BuildSettings {
 	}
 
 	/// Attempts to determine the SDK this scheme builds for.
-	public var buildSDK: Result<SDK, CarthageError> {
+	public var firstBuildSDK: Result<SDK, CarthageError> {
 		return self["PLATFORM_NAME"].flatMap(SDK.fromString)
 	}
 
+	public var buildSDK: SignalProducer<SDK, CarthageError> {
+		let supportedPlatforms = self["SUPPORTED_PLATFORMS"]
+
+		if let supportedPlatforms = supportedPlatforms.value {
+			let platforms = split(supportedPlatforms) { $0 == " " }
+			return SignalProducer<String, CarthageError>(values: platforms)
+				|> map { platform in SignalProducer(result: SDK.fromString(platform)) }
+				|> flatten(.Merge)
+		}
+
+		return SignalProducer(result: firstBuildSDK)
```

Comment:
I think you could get rid of `firstBuildSDK` and just incorporate that logic here?